src/send/compression.rs

```diff
@@ -37,110 +13,83 @@ pub(crate) fn bytes2type(bytes: [u8; 5]) -> Format {
     }
 }
 
-cfg_if! {
-    if #[cfg(feature = "gz")] {
-        pub(crate) fn new_gz_encoder<'a>(out: Box<dyn io::Write + Send + 'a>, level: Level) -> Result<Box<dyn io::Write + Send + 'a>, Error> {
-            Ok(Box::new(flate2::write::GzEncoder::new(
-        out,
-        level.into(),
-            )))
-        }
-
-        pub(crate) fn new_gz_decoder<'a>(
-            inp: Box<dyn io::Read + Send + 'a>,
-        ) -> Result<(Box<dyn io::Read + Send + 'a>, Format), Error> {
-            Ok((
-        Box::new(flate2::read::MultiGzDecoder::new(inp)),
-        Format::Gzip,
-            ))
-        }
-    } else {
-        pub(crate) fn new_gz_encoder<'a>(_: Box<dyn io::Write + Send + 'a>, _: Level) -> Result<Box<dyn io::Write + Send + 'a>, Error> {
-            Err(Error::FeatureDisabled)
-        }
-
-        pub(crate) fn new_gz_decoder<'a>(_: Box<dyn io::Read + Send + 'a>) -> Result<(Box<dyn io::Read + Send + 'a>, Format), Error> {
-            Err(Error::FeatureDisabled)
-        }
-    }
-}
-
-cfg_if! {
-    if #[cfg(feature = "bz2")] {
-        pub(crate) fn new_bz2_encoder<'a>(out: Box<dyn io::Write + Send + 'a>, level: Level) -> Result<Box<dyn io::Write + Send + 'a>, Error> {
-            Ok(Box::new(bzip2::write::BzEncoder::new(
-                out,
-                level.into(),
-            )))
-        }
-
-        pub(crate) fn new_bz2_decoder<'a>(
-            inp: Box<dyn io::Read + Send + 'a>,
-        ) -> Result<(Box<dyn io::Read + Send + 'a>, Format), Error> {
-            Ok((
-                Box::new(bzip2::read::BzDecoder::new(inp)),
-                Format::Bzip,
-            ))
-        }
-    } else {
-        pub(crate) fn new_bz2_encoder<'a>(_: Box<dyn io::Write + Send + 'a>, _: Level) -> Result<Box<dyn io::Write + Send + 'a>, Error> {
-            Err(Error::FeatureDisabled)
-        }
-
-        pub(crate) fn new_bz2_decoder<'a>(_: Box<dyn io::Read + Send + 'a>) -> Result<(Box<dyn io::Read + Send + 'a>, Format), Error> {
-            Err(Error::FeatureDisabled)
-        }
-    }
-}
-
-cfg_if! {
-    if #[cfg(feature = "lzma")] {
-    pub(crate) fn new_lzma_encoder<'a>(out: Box<dyn io::Write + Send + 'a>, level: Level) -> Result<Box<dyn io::Write + Send + 'a>, Error> {
-            Ok(Box::new(liblzma::write::XzEncoder::new(out, level.into())))
+impl_format!(
+    gz,
+    "gz",
```

Comment:
These two are a bit repetitive, need to figure out how to make an [ident](https://veykril.github.io/tlborm/decl-macros/minutiae/fragment-specifiers.html#ident) into a [literal](https://veykril.github.io/tlborm/decl-macros/minutiae/fragment-specifiers.html#literal) in the macro